### ISSUE TITLE
all: decide how to spell the past tense of cancel

### ISSUE THREAD

**Issue by robpike** [golang/go#11626]
Both the gerund and the past participle of cancel appear with both one and with two l's. Be consistent and also correct. Check the dictionary.

**Comment by sprstnd** [comment]
Discourage the use of inflection.
cancel
cancelled
cancelling
cancellers
cancellation
etc...

**Comment by sprstnd** [comment]
Just an interesting data set.
https://books.google.com/ngrams/graph?content=canceled%2Ccancelled&year_start=1800&year_end=2010&corpus=5&smoothing=3&share=&direct_url=t1%3B%2Ccanceled%3B%2Cc0%3B.t1%3B%2Ccancelled%3B%2Cc0

**Comment by robpike** [comment]
The key phrase in the issue opener was "be consistent".

**Comment by sprstnd** [comment]
And that is why I suggested to discourage inflection, as cancellation doesn't seem to have an inflected form. All suffix forms of cancel would be consistent, with two l's. cancel, cancelled, cancelling, cancellation.

But you also applied that one style was incorrect too- within that same sentence. And that is where I corrected you- both styles are correct and acceptable in both American English and British English.

**Comment by theckman** [comment]
Looking at Google search results for `cancelation` makes it seem widespread enough that it should be considered valid. If you even ask the engine to define it, it provides a valid definition. Based on that, I have no objections with choosing the single `l` form.

**Comment by robpike** [comment]
Whatever you think I implied, what I said was, "check the dictionary". I wanted to be sure to be correct. Google searches are good at finding usage, and there are arguments that common usage is always correct, but if it's easy to obey what the dictionary claims is true, why not do that?

**Comment by mwhudson** [comment]
Which dictionary do you mean though? The OED only has "cancelled", Mirriam-Webster says both "cancelled" and "canceled" are OK. My understanding is that "canceled" is predominant but not universal in the USA and not used in other English speaking countries (I find canceled looks wrong, but I grew up in the UK and live in NZ so that's not surprising).

**Comment by robpike** [comment]
mwhudson: See issue #11626. :)

**Comment by sprstnd** [comment]
I tried to make this clear as possible. Stating that both forms are ACCEPTAbLE and CORRECT in both American English and British English. I am suggesting discourage inflection and use the long form, as cancellation doesn't seem to have an accepted inflected variant. And I am not talking about google searches of web pages, blogs, and such- I am strictly speaking about academia. As in my previous posts, I suggest using the long form so that all the suffixed variants of cancel are consistent. - cancel, cancelling, cancelled, cancellation.

There are arguments for each side. I stated my opinion- not much else to argue. It boils down to the nitty gritty of taste. robpike will do what robpike does, its your task commander assignee.

Interesting data sets can be produced using Ngrams: https://books.google.com/ngrams
Maybe experiment before making your decision.

**Comment by sprstnd** [comment]
what you said was, "Both the gerund and the past participle of cancel appear with both one and with two l's. Be consistent and also correct. Check the dictionary." That statement implied that the use of suffixed forms of cancel were used in BOTH an inconsistent and incorrect manner. I merely corrected you by stating that both are acceptable, and now I'll add that, its a matter of taste. I am sorry I bruised your ego, it was not my intention to turn this into a pissing match.

**Comment by sprstnd** [comment]
theckman suggested the use of short form. stating that cancelation seemed to be widely used. However this data set suggest otherwise:
https://books.google.com/ngrams/graph?content=cancellation%2Ccancelation&year_start=1800&year_end=2008&corpus=15&smoothing=0&share=&direct_url=t1%3B%2Ccancellation%3B%2Cc0%3B.t1%3B%2Ccancelation%3B%2Cc0

**Comment by cespare** [comment]
The OED only recognizes 'cancelled' which is the preferred spelling everywhere that isn't the US.

All the other dictionaries I looked at (Merriam-Webster, American Heritage, wiktionary) list both 'canceled' and 'cancelled'.

The ngram results discussed above as well is lots of other online sources suggest that 'canceled' is more prevalent in American usage and 'cancelled' is nigh-universal elsewhere.

The AP style guide recommends 'canceled'.

I'd say go with 'canceled':

* Dictionaries support either choice
* The Go documentation prefers American spellings already (exhibit A: image/color)
* It is one character shorter than the alternative

**Comment by theckman** [comment]
@cespare I hadn't even thought to check other uses of words like that in the codebase. :+1: on 'canceled'

**Comment by sprstnd** [comment]
I'd still like to point out that cancellation doesn't seem to have an inflicted variant. using the long form keeps all suffixed forms consistent. As these forms could used in future documentation it may be prudent decision making.

cancel, cancelling,cancelled, cancellation

Also, one could make the argument that all forms of cancel, inflicted or not, are so widely accepted that this nitpicking is futile.

I made the argument 

cancel, cancelling, cancelled, cancellation

**Comment by theckman** [comment]
As I understand it `cancelation` is perfectly valid in respect to Americanized English. And we already use American spellings according to @cespare. Seems reasonable to me.

**Comment by sprstnd** [comment]
I've already made the argument that it is valid(accepted), however I also pointed out that its not widely used like the other inflicted suffixed variants of cancel.

But maybe this project can be a turning point for the use of cancelation.

just a side note. my browsers spell checking doesn't recognized 'cancelation', however it accepts all other inflicted suffixed variants of cancel. I'll check other spell checking software for the same behavior.

Like I said before, it is a matter of taste, I'm just making the argument for commonality among existing American/British English work.
